lib/matobjflist.gi

```diff
@@ -0,0 +1,512 @@
+            # set the row-th wor of the product to w.
+            l{[(row-1)*b![FLISTREP_NCPOS]+1..row*b![FLISTREP_NCPOS]]} := w;
+        fi;
+    od;
+    if not IsMutable(a) and not IsMutable(b) then
+        MakeImmutable(l);
+    fi;
+    return Objectify( ty, [a![FLISTREP_BDPOS],a![FLISTREP_NRPOS],b![FLISTREP_NCPOS],l] );
+  end );
+
+  InstallMethod( ConstructingFilter, "for an flist matrix",
+  [ IsFlistMatrixRep ],
+  function( mat )
+    return IsFlistMatrixRep;
+  end );
+
+  InstallMethod( \=, "for two flist matrices",
+  [ IsFlistMatrixRep, IsFlistMatrixRep ],
+  function( a, b )
+    return a![FLISTREP_BDPOS] = b![FLISTREP_BDPOS] and a![FLISTREP_NCPOS] = b![FLISTREP_NCPOS] and a![FLISTREP_NRPOS] = b![FLISTREP_NRPOS] and EQ_LIST_LIST_DEFAULT(a![FLISTREP_ELSPOS],b![FLISTREP_ELSPOS]);
+  end );
+
+  InstallMethod( \<, "for two flist matrices",
+  [ IsFlistMatrixRep, IsFlistMatrixRep ],
+  function( a, b )
+    return LT_LIST_LIST_DEFAULT(a![FLISTREP_ELSPOS],b![FLISTREP_ELSPOS]);
+  end );
+
+  # The following methods need not ne implemented for any specific MatrixRep as
+  # default methods for all MatrixObjects exist
+  InstallMethod( AdditiveInverseSameMutability, "for an flist matrix",
+  [ IsFlistMatrixRep ],
+  function( mat )
+    local l;
+    l := List(mat![FLISTREP_ELSPOS],AdditiveInverseSameMutability);
+    if not IsMutable(mat) then
+        MakeImmutable(l);
+    fi;
+    return Objectify( TypeObj(mat), [mat![FLISTREP_BDPOS],mat![FLISTREP_NRPOS],mat![FLISTREP_NCPOS],l] );
+  end );
+
+  InstallMethod( AdditiveInverseImmutable, "for an flist matrix",
+  [ IsFlistMatrixRep ],
+  function( mat )
+    local l,res;
+    l := List(mat![FLISTREP_ELSPOS],AdditiveInverseImmutable);
+    res := Objectify( TypeObj(mat), [mat![FLISTREP_BDPOS],mat![FLISTREP_NRPOS],mat![FLISTREP_NCPOS],l] );
+    MakeImmutable(res);
+    return res;
+  end );
+
+InstallMethod( AdditiveInverseMutable, "for an flist matrix",
+  [ IsFlistMatrixRep ],
+  function( mat )
+    local l,res;
+    l := List(mat![FLISTREP_ELSPOS],AdditiveInverseMutable);
+    res := Objectify( TypeObj(mat), [mat![FLISTREP_BDPOS],mat![FLISTREP_NRPOS],mat![FLISTREP_NCPOS],l] );
+    if not IsMutable(mat) then
+        SetFilterObj(res,IsMutable);
+    fi;
+    return res;
+  end );
+
+  # Further methods one could implement if it offers a significant performance
+  # improvement
+  # ZeroMutable
+  # IsZero
+  # IsOne
+  # OneSameMutability
+  # OneMutable
+  # OneImmutable 
+
+  # This method should be implemented with care as e.g. for permutation
+  # matrices, diagonal matrices or similar inversion can be achieved cheaply.
+  # For sparse matrices on the other hand one must avoid creating large
+  # intermediary objects. In this particular implementation the existing method
+  # from GAP is called.
+
+  InstallMethod( InverseMutable, "for an flist matrix",
+  [ IsFlistMatrixRep ],
+  function( mat )
+    local n;
+    if m![FLISTREP_NCPOS] <> mat![FLISTREP_NRPOS] then
```

Comment:
you probably mean mat![FLISTREP_NCPOS] <> mat![FLISTREP_NRPOS]

Reply:
oh yes thank you!